sync-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java

```diff
@@ -128,6 +135,11 @@ public void run() {
             }
         });
         queue.shutdown();
+        try {
+            queue.awaitTermination(200, TimeUnit.SECONDS);
```

Comment:
Why bother waiting here? I'm not sure what it add to it.

Really I think you want something which calls `shutdownNow()` once `db.close()` has been called. Can you call it from the task itself?